### PR TITLE
[Method References 2] 165169 Convert method reference to ServiceTrackerList in module portal-search | part 1

### PR DESCRIPTION
**Pull request by W2jx**
Convert method reference to ServiceTrackerList in module portal-search.

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:search

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/SearchPermissionCheckerImpl.java [hunk 1]
```diff
@@ -622,10 +599,10 @@ private String _getPermissionName(
 	private static final Log _log = LogFactoryUtil.getLog(
 		SearchPermissionCheckerImpl.class);
 
-	private final Collection<SearchPermissionFieldContributor>
-		_searchPermissionFieldContributors = new CopyOnWriteArrayList<>();
-	private final Collection<SearchPermissionFilterContributor>
-		_searchPermissionFilterContributors = new CopyOnWriteArrayList<>();
+	private volatile ServiceTrackerList<SearchPermissionFieldContributor>
```
Comment (tinatian) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/SearchPermissionCheckerImpl.java]: @W2jx Why it needs to be volatile?

**Comment by tinatian** [comment]
@W2jx I did not know what you inlined in this commit https://github.com/liferay-core-infra/liferay-portal/pull/2278/commits/354a81eeea71bfc83e2218344e1f02b8082d1996.

**Comment by tinatian** [comment]
@W2jx Please resend after you did what Brian requested at https://github.com/brianchandotcom/liferay-portal/pull/126022#issuecomment-1309426896

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/SearchPermissionCheckerImpl.java (hunk 1 above)
Comment (W2jx) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/SearchPermissionCheckerImpl.java]: Hi @tinatian , SF added it automatically.

**Pull request closed without merging**